src/chunkserver/network_worker_thread.cc

```diff
@@ -18,1523 +18,51 @@
-			bytesRead =
-			    ::read(sock, inputPacket.startPtr, inputPacket.bytesLeft);
-			if (bytesRead == 0) {
-				state = State::Close;
-				return;
-			}
-			if (bytesRead < 0) {
-				if (errno != EAGAIN) {
-					safs_silent_errlog(LOG_NOTICE, "(read) read error");
-					state = State::Close;
-				}
-				return;
-			}
-			stats_bytesin += bytesRead;
-			inputPacket.startPtr += bytesRead;
-			inputPacket.bytesLeft -= bytesRead;
-
-			if (inputPacket.bytesLeft > 0) {
-				return;
-			}
-		}
-		if (writeJobId == 0) {
-			ptr = headerBuffer;
-			type = get32bit(&ptr);
-			opSize = get32bit(&ptr);
-
-			mode = Mode::Header;
-			inputPacket.bytesLeft = PacketHeader::kSize;
-			inputPacket.startPtr = headerBuffer;
-
-			gotPacket(type, inputPacket.packet.data(), opSize);
-		}
-	}
-}
-
-void ChunkserverEntry::writeToSocket() {
-	TRACETHIS();
-	PacketStruct *pack = nullptr;
-	int32_t bytesWritten;
-
-	for (;;) {
-		if (outputPackets.empty()) { return; }
-
-		pack = outputPackets.front().get();
-
-		if (pack->outputBuffer) {
-			size_t bytesInBufferBefore = pack->outputBuffer->bytesInABuffer();
-			OutputBuffer::WriteStatus ret =
-			    pack->outputBuffer->writeOutToAFileDescriptor(sock);
-			size_t bytesInBufferAfter = pack->outputBuffer->bytesInABuffer();
-			massert(bytesInBufferAfter <= bytesInBufferBefore,
-					"New bytes in pack->outputBuffer after sending some data");
-			stats_bytesout += (bytesInBufferBefore - bytesInBufferAfter);
-			if (ret == OutputBuffer::WRITE_ERROR) {
-				safs_silent_errlog(LOG_NOTICE, "(write) write error");
-				state = State::Close;
-				return;
-			} else if (ret == OutputBuffer::WRITE_AGAIN) {
-				return;
-			}
-		} else {
-			bytesWritten = ::write(sock, pack->startPtr, pack->bytesLeft);
-			if (bytesWritten == 0) {
-				state = State::Close;
-				return;
-			}
-			if (bytesWritten < 0) {
-				if (errno != EAGAIN) {
-					safs_silent_errlog(LOG_NOTICE, "(write) write error");
-					state = State::Close;
-				}
-				return;
-			}
-			stats_bytesout += bytesWritten;
-			pack->startPtr += bytesWritten;
-			pack->bytesLeft -= bytesWritten;
-			if (pack->bytesLeft > 0) {
-				return;
-			}
-		}
-		// packet has been sent
-		if (pack->outputBuffer) {
-			getReadOutputBufferPool().put(std::move(pack->outputBuffer));
-		}
-		outputPackets.pop_front();
-		outputCheckReadFinished();
-	}
+constexpr int getConnectTimeout(int cnt) {
+	return cnt % 2 ? kTimeoutOdd * (1 << (cnt >> 1))
+	               : kTimeoutEven * (1 << (cnt >> 1));
 }
 
-NetworkWorkerThread::NetworkWorkerThread(uint32_t nrOfBgjobsWorkers, uint32_t bgjobsCount)
-		: doTerminate(false) {
+NetworkWorkerThread::NetworkWorkerThread(uint32_t nrOfBgjobsWorkers,
+                                         uint32_t bgjobsCount)
+    : doTerminate(false) {
 	TRACETHIS();
 	eassert(pipe(notify_pipe) != -1);
 #ifdef F_SETPIPE_SZ
-	eassert(fcntl(notify_pipe[1], F_SETPIPE_SZ, 4096*32));
+	eassert(fcntl(notify_pipe[1], F_SETPIPE_SZ, 4096 * 32));
```

Comment:
Consider adding a descriptive naming for magic constant `4096 * 32`

Reply:
Not in the scope of this change.
will be handled as part of the refactoring, but this change is only moving to new files to facilitate the review.